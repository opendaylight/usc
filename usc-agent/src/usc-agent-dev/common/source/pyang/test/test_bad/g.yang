module g {
    namespace "urn:g";
    prefix g;
    
    grouping ww {
        list a {
            leaf b { type int32; }
        }
    }

    container f {
        uses ww; // error: list a needs a key
        leaf bar {
            type leafref {
                path "../h/b"; // error: b not key
            }
        }
        list h {
            key a;
            leaf a {
                type int32;
            }
            leaf b {
                type int32;
            }
        }
    }

    rpc foo {
        input d { // error: bad argument
            container x;
        }
    }
    augment "/foo/d/x" { // error: d/x not found
        leaf a {
            type int32;
        }
    }

    rpc xfoo {
        typedef a1 {
            type uint32;
        }
        grouping a2 {
            leaf x {
                type a1;
                default "-1"; // error: bad default
            }
        }
        typedef a3 {
            type a1 {
                range "10..max";
            }
        }
        grouping a4 {
            leaf y {
                type a3;
                default 1; // error: bad default
            }
        }
        input {
            uses a4 {
                refine y {
                    default 2; // error: bad default
                }
            }
        }
        input { // error: multiple input
            uses a2;
        }
    }

  list xx {
    key f:a;
    leaf a {
      type int32;
    }
  }

}