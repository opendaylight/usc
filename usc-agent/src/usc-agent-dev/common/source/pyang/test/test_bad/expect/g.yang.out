g.yang:12: error: NEED_KEY_USES
g.yang:30: error: UNEXPECTED_ARGUMENT
g.yang:34: error: NODE_NOT_FOUND
g.yang:34: error: NODE_NOT_FOUND
g.yang:44: warning: UNUSED_GROUPING
g.yang:47: error: TYPE_VALUE
g.yang:58: error: TYPE_VALUE
g.yang:64: error: TYPE_VALUE
g.yang:68: error: UNEXPECTED_KEYWORD
g.yang:68: error: DUPLICATE_CHILD_NAME
g.yang:74: error: BAD_KEY
